src/compiler/checker.ts

```diff
@@ -26548,8 +26572,9 @@ namespace ts {
          */
         function checkPropertyAccessibility(
             node: PropertyAccessExpression | QualifiedName | PropertyAccessExpression | VariableDeclaration | ParameterDeclaration | ImportTypeNode | PropertyAssignment | ShorthandPropertyAssignment | BindingElement,
-            isSuper: boolean, type: Type, prop: Symbol, reportError = true): boolean {
-            const flags = getDeclarationModifierFlagsFromSymbol(prop);
+            isSuper: boolean, type: Type, prop: Symbol, reportError = true, isWrite = false): boolean {

```

Comment:
```suggestion
            isSuper: boolean, isWrite: boolean, type: Type, prop: Symbol, reportError = true): boolean {
```

1. checkPropertyAccessibility is only called four times.
2. A string of boolean parameters is error-prone, a string of trailing optional booleans is worse.
3. isWrite is the same category as isSuper.

Reply:
👍

In my defense, `reportError` wasn't there when I wrote this code 😅